File: rust/muse/src/algorithms/sorting/insertion_sort.rs
pub fn sort(arr: &mut [i32]) {
    for i in 1..arr.len() {
        let target = arr[i];
        let mut cursor = i;
        while cursor > 0 {
            if arr[cursor - 1] <= target {
                break;
            }
            arr[cursor] = arr[cursor - 1];
            cursor -= 1;
        }
        arr[cursor] = target;
    }
}

#[cfg(test)]
mod tests;

File: python/muse/algorithms/sorting/insertion_sort.py
def sort(arr: list[int]) -> None:
    for i in range(1, len(arr)):
        target = arr[i]
        for cursor in range(i, 0, -1):
            if arr[cursor - 1] <= target:
                break
            arr[cursor] = arr[cursor - 1]
        else:
            cursor = 0
        arr[cursor] = target
